homeassistant/components/opentherm_gw/climate.py

```diff
@@ -79,7 +86,8 @@ def __init__(self, gw_dev, options):
     def update_options(self, entry):
         """Update climate entity options."""
         self.floor_temp = entry.options[CONF_FLOOR_TEMP]
-        self.temp_precision = entry.options[CONF_PRECISION]
+        self.temp_read_precision = entry.options[CONF_READ_PRECISION]
```

Comment:
Have we released a version of the integration with the `CONF_PRECISION` in the options? Then we need to convert that item to the new options data schema.

Reply:
CONF_PRECISION was the name of the option before I divided it into read and set precision. CONF_PRECISION was configured in the options flow schema. But I replaced it with CONF_READ_PRECISION. CONF_PRECISION only still exists because it was also in the yaml config schema and I had to revert that to it's original state.

Reply:
Ok. How should we migrate existing users that had set `CONF_PRECISION`?

Reply:
Hmm.. good point. I like how well you're thinking this through. 

What I could do is set the const `CONF_READ_PRECISION = read_precision` back to `CONF_READ_PRECISION = precision` so it will use its original setting. Then I could make `CONF_SET_PRECISION` mirror `CONF_READ_PRECISION` when it is not set. 

Would you be ok with that?

Reply:
I think we can keep the new term named `"read_precision"` since it's more clear what that is now. We can just add a check for the old term too, `"precision"` if `"read_precision"` is not set in entry.options, and use the `"precision"` value for read precision in that case.

Sounds good to default the set precision to the same as the read precision.

Reply:
After some rethinking I am a bit in doubt. The original `"precision"` setting won't be available from the options form anymore, so when the new settings aren't used it won't be possible to get back to the default setting if `"precision"` was set before. Which somehow forces you to use the new settings.

Would it be possible to set `"read_precision"` to the value of `"precision"` and then set `"precision"` to None? So you won't be bothered by the 'old' setting anymore.

Something like this in the `__init__` or `update_options` functions, or in another function if that would be better?
```
if entry.options[CONF_PRECISION]:
    entry.options[CONF_READ_PRECISION] = entry.options[CONF_PRECISION] 
    entry.options[CONF_PRECISION] = None
self.temp_read_precision = entry.options[CONF_READ_PRECISION]
```
Is something like this possible at all? Or would it be better to use `"precision"` instead of `"read_precision"` after all?
Or am I overthinking stuff right now?